src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -43,9 +44,350 @@ Copyright (c) 2013, Ford Motor Company
+        } else {
+          LOG4CXX_WARN(logger_,
+                       kQueryAppsValidationFailedPrefix
+                           << "ttsName is not the string type.");
+          return false;
+        }
+        if (!language[*iter].keyExists(json::vrSynonyms)) {
+          LOG4CXX_WARN(logger_,
+                       kQueryAppsValidationFailedPrefix
+                           << "'languages.vrSynonyms' doesn't exist");
+          return false;
+        }
+        const smart_objects::SmartArray* synonyms_array =
+            language[*iter][json::vrSynonyms].asArray();
+        if (!synonyms_array) {
+          LOG4CXX_WARN(logger_,
+                       kQueryAppsValidationFailedPrefix
+                           << "vrSynonyms is not array.");
+          return false;
+        }
+        const size_t synonyms_array_size = synonyms_array->size();
+        if (synonyms_array_size < kVrArraySizeMin) {
+          LOG4CXX_WARN(logger_,
+                       kQueryAppsValidationFailedPrefix
+                           << "vrSynomyms array has [" << synonyms_array_size
+                           << "] size < allowed min size [" << kVrArraySizeMin
+                           << "]");
+          return false;
+        }
+        if (synonyms_array_size > kVrArraySizeMax) {
+          LOG4CXX_WARN(logger_,
+                       kQueryAppsValidationFailedPrefix
+                           << "vrSynomyms array size [" << synonyms_array_size
+                           << "] exceeds maximum allowed size ["
+                           << kVrArraySizeMax << "]");
+          return false;
+        }
+        for (std::size_t idx = 0; idx < synonyms_array_size; ++idx) {
+          const smart_objects::SmartObject& synonym = (*synonyms_array)[idx];
+          const std::string vrSynonym = synonym.asString();
+          if (vrSynonym.length() > kVrSynonymLengthMax) {
+            LOG4CXX_WARN(logger_,
+                         kQueryAppsValidationFailedPrefix
+                             << "vrSYnomym item [" << idx
+                             << "] exceeds max length [" << vrSynonym.length()
+                             << "]>[" << kVrSynonymLengthMax << "]");
+            return false;
+          }
+          if (vrSynonym.length() < kVrSynonymLengthMin) {
+            LOG4CXX_WARN(logger_,
+                         kQueryAppsValidationFailedPrefix
+                             << "vrSYnomym item [" << idx << "] length ["
+                             << vrSynonym.length()
+                             << "] is less then min length ["
+                             << kVrSynonymLengthMin << "] allowed.");
+            return false;
+          }
+          // Verify duplicates
+          SynonymsMap::iterator synonyms_map_iter = synonyms_map.find(*iter);
+          SynonymsSet* synonyms_set = &(synonyms_map_iter->second);
+          if (synonyms_map_iter != synonyms_map.end()) {
+            if ((*synonyms_map_iter).second.find(vrSynonym) ==
```

Comment:
@okoreniak, you don't have to check set for key existence. In case of key already present in set `insert()` method returns false